### PR TITLE
[BEAM-12898] Flink Load Tests failure- UncheckedExecutionException - leaking vms

### PR DESCRIPTION
**Pull request by andoni-guzman**
- Fixed dataproc cluster creation. Dataproc is now in 2.0 version
- Flink 1.12, Used this version because dataproc 2.0 supports flink 1.12
- Fixed leaked vm's 

This tests still disabled, the reason is an error related with a Class **Encountered unsupported logical type URN: int01:43:59 2021/09/14 08:43:59  (): org.apache.beam.vendor.guava.v26_0_jre.com.google.common.util.concurrent.UncheckedExecutionException ** 

  -- beam_LoadTests_Go_CoGBK_Flink_batch 
  -- beam_LoadTests_Go_Combine_Flink_Batch
  -- beam_LoadTests_Go_GBK_Flink_Batch 

This tests still disabled, the reason is an error related with a Class **java.lang.NoClassDefFoundError: Could not initialize class org.apache.beam.runners.core.construction.SerializablePipelineOptions ** 
This problem affects Go and Python tests ( Not all of them) so I think it's a problem related with the runner and not with a language and was reported time ago https://issues.apache.org/jira/browse/BEAM-11753 

The error seems to be related with dependencies , however adding them it's not a good solution according to https://github.com/apache/beam/pull/14953 , so the plan is enable this tests, create other jira and enabled the follow test when the problem will fixed
 Tests still disabled 
  -- beam_LoadTests_Go_ParDo_Flink_Batch (FAILED) 
  -- beam_LoadTests_Go_SideInput_Flink_Batch (FAILED)
  -- beam_LoadTests_Python_Combine_Flink_Batch (FAILED)
  -- beam_LoadTests_Python_Combine_Flink_Streaming (FAILED)
  -- beam_LoadTests_Python_ParDo_Flink_Batch (FAILED)
  -- beam_LoadTests_Python_ParDo_Flink_Streaming (FAILED)------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by riteshghorse** [comment]
assign to next reviewer

**Comment by andoni-guzman** [comment]
R: @ibzib 
HI Kyle, can you review my PR or assing someone to do it please?
Thanks a lot

**Comment by ibzib** [comment]
Run Load Tests Go GBK Flink Batch

**Comment by ibzib** [comment]
Run Load Tests Go Combine Flink Batch

File: sdks/go/pkg/beam/core/runtime/graphx/translate.go [hunk 1]
```diff
@@ -36,7 +36,7 @@ import (
 // TODO(lostluck): 2018/05/28 Extract these from their enum descriptors in the pipeline_v1 proto
 const (
 	URNImpulse       = "beam:transform:impulse:v1"
-	URNParDo         = "beam:transform:pardo:v1"
+	URNParDo         = "beam:transform:org.apache.beam:pardo:v1"
```
Comment (ibzib) [inline sdks/go/pkg/beam/core/runtime/graphx/translate.go:39]: I don't think this is a correct URN. These URNs need to match the definitions here: https://github.com/apache/beam/blob/5ef9b7eb433e448b7f0fa37485623df0de6a8eef/model/pipeline/src/main/proto/beam_runner_api.proto#L224

**Review by ibzib — COMMENTED** [review]
@andoni-guzman looks like some tests are still failing.

File: sdks/go/pkg/beam/core/runtime/graphx/translate.go (hunk 1 above)
Comment (andoni-guzman) [reply sdks/go/pkg/beam/core/runtime/graphx/translate.go:39]: Hi @ibzib, the reason that I did in that way is based in this PR https://github.com/apache/beam/pull/15748/files 
This PR Updates cross-language transform URNs to use the new convention and solve "Encountered unsupported logical type URN" I don't know why in my local tests I didn't catch the new failure.
If I use the URN "beam:transform:pardo:v1" I got logical type URN error.
What is the best aprox for this? 
Maybe only fix dataproc creation and deletion and in a future solve each flink tests?
Thanks

Comment (lostluck) [reply sdks/go/pkg/beam/core/runtime/graphx/translate.go:39]: Agreed with Kyle. The Proto is Canonical, and that PR didn't change the canonical URN for all ParDos. 

The referenced PR was specifically for expansion requests for Cross Language transforms, not DoFns executing in their native environment, which is what this URN represents.

What is the actual bug for "unsupported logical type" that your'e getting? That's how it's reporting it, but is there more to the exception than that?

Comment (andoni-guzman) [reply sdks/go/pkg/beam/core/runtime/graphx/translate.go]: Hi @ibzib I disable the tests that failed and back to the original URN. I think this PR is ready to merge.
A question related with the tracking of the  disabled tests. It's better create new jiras for each error ( at this moment there is two error that affect the tests) and close BEAM-12898 Leaking Vm's or still had open  beam 12898 with a note in the comments that dataproc creation,deletion and fixed vm's is solved and only the tests are missing?

File: .test-infra/dataproc/flink_cluster.sh [hunk 2]
```diff
@@ -133,7 +134,8 @@ function create_cluster() {
 
   # Docker init action restarts yarn so we need to start yarn session after this restart happens.
   # This is why flink init action is invoked last.
-  gcloud dataproc clusters create $CLUSTER_NAME --region=global --num-workers=$num_dataproc_workers --initialization-actions $DOCKER_INIT,$BEAM_INIT,$FLINK_INIT --metadata "${metadata}", --image-version=$image_version --zone=$GCLOUD_ZONE --quiet
+  #--initialization-actions $DOCKER_INIT,$BEAM_INIT,$FLINK_INIT Older initialization
```
Comment (ibzib) [inline .test-infra/dataproc/flink_cluster.sh:137]: I don't think we need any of these comments anymore.

File: .test-infra/jenkins/job_LoadTests_Combine_Flink_Go.groovy [hunk 3]
```diff
@@ -133,5 +133,5 @@ CronJobBuilder.cronJob('beam_LoadTests_Go_Combine_Flink_Batch', 'H 8 * * *', thi
     influx_hostname: InfluxDBCredentialsHelper.InfluxDBHostUrl,
   ]
   // TODO(BEAM-12898): Re-enable this test once fixed.
-  // loadTestJob(delegate, CommonTestProperties.TriggeringContext.POST_COMMIT, 'batch')
+  //loadTestJob(delegate, CommonTestProperties.TriggeringContext.POST_COMMIT, 'batch')
```
Comment (ibzib) [inline .test-infra/jenkins/job_LoadTests_Combine_Flink_Go.groovy:136]: We can re-enable these. It's fine if they're failing as long as we're not leaking VMs (which I believe was fixed in #15547).

File: .test-infra/jenkins/CommonTestProperties.groovy [hunk 4]
```diff
@@ -26,7 +26,7 @@ class CommonTestProperties {
   }
 
   static String getFlinkVersion() {
-    return "1.13"
+    return "1.12" //Flink Version in dataproc 2.0
```
Comment (ibzib) [inline .test-infra/jenkins/CommonTestProperties.groovy:29]: I don't think this sets the Flink version used in the load tests. I think that's set in each load test config, e.g.

https://github.com/apache/beam/blob/d3f320426a115b8c986a817fe2ba87f9fd7f2192/.test-infra/jenkins/job_LoadTests_GBK_Flink_Go.groovy#L201

Looks like it's already set to 1.12, so you can probably revert this change.

**Review by ibzib — CHANGES_REQUESTED** [review]

File: sdks/go/pkg/beam/core/runtime/graphx/translate.go (hunk 1 above)
Comment (ibzib) [reply sdks/go/pkg/beam/core/runtime/graphx/translate.go]: After this is merged, we can close BEAM-12898 and create new issues for the test failures.

**Comment by ibzib** [comment]
Run Load Tests Go GBK Flink Batch

**Comment by ibzib** [comment]
Run Load Tests Go GBK Flink Batch

**Comment by ibzib** [comment]
Looks like the test "failed successfully" and deleted the cluster after it was done. LGTM
